test/Microsoft.AspNet.Mvc.TagHelpers.Test/TagHelperOutputExtensionsTest.cs

```diff
@@ -80,6 +80,39 @@ public void CopyHtmlAttribute_DoesNotOverrideAttributes()
         }
 
         [Fact]
+        public void CopyHtmlAttribute_ThrowsWhenUnknownAttribute()
+        {
+            // Arrange
+            var invalidAttributeName = "hello2";
+            var tagHelperOutput = new TagHelperOutput(
+                "p",
+                attributes: new Dictionary<string, object>());
+            var tagHelperContext = new TagHelperContext(
+                allAttributes: new Dictionary<string, object>(StringComparer.Ordinal)
+                {
+                    { "hello", "world" }
+                },
+                items: new Dictionary<object, object>(),
+                uniqueId: "test",
+                getChildContentAsync: () =>
+                {
+                    var tagHelperContent = new DefaultTagHelperContent();
+                    tagHelperContent.Append("Something");
+                    return Task.FromResult<TagHelperContent>(tagHelperContent);
+                });
+
+            string expectedMessage = 
+                "The attribute 'hello2' does not exist in the TagHelperContext." + Environment.NewLine +
+                "Parameter name: attributeName";
```

Comment:
if you prefer, can use a helper (from the `Microsoft.AspNet.Testing` namespace) instead:
``` c#
ExceptionAssert.ThrowsArgument(
  () => tagHelperOutput.CopyHtmlAttribute(invalidAttributeName, tagHelperContext),
  "attributeName",
  "The attribute 'hello2' does not exist in the TagHelperContext.");
```

then would need neither `expectedMessage` nor `ex` variables.